### PR TITLE
Total assessment value adding

### PR DESCRIPTION
**Pull request by Space-tourist**

### CODE REVIEW

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AcceptEstimate/Services/AcceptEstimateCommandCreator.cs [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
 internal sealed class AcceptEstimateCommandCreator : ICommandCreator
 {
-    public string Command => CommandList.AcceptEstimate;
+    private readonly AssessmentValue.Value _value;
+    public string Command { get; }
     public bool SupportSingleLineMode => false;
+
+    public AcceptEstimateCommandCreator(string command, AssessmentValue.Value value)
+    {
+        if (string.IsNullOrWhiteSpace(command))
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AcceptEstimate/Services/AcceptEstimateCommandCreator.cs:15]: string.IsNullOrWhiteSpace(command) -> ArgumentException.ThrowIfNullOrWhiteSpace(command)

File: src/Inc.TeamAssistant.Appraiser.DataAccess/StoryReader.cs [hunk 2]
```diff
@@ -25,11 +25,16 @@ public StoryReader(IConnectionFactory connectionFactory)
...
+                WHEN s.story_type = 1
+                    THEN SUM(s.total_value)
+                    ELSE COUNT(*)
+                END as assessmentsum
             FROM appraiser.stories AS s
             WHERE s.team_id = @team_id AND s.created <= @before
             AND (@from is null OR s.created >= @from)
-            GROUP BY DATE(s.created)
+            GROUP BY DATE(s.created), s.story_type
             ORDER BY DATE(s.created) DESC;",
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.DataAccess/StoryReader.cs:38]: Please add s.story_type to ORDER BY

File: src/Inc.TeamAssistant.Appraiser.Domain/Story.cs [hunk 3]
```diff
@@ -193,4 +194,16 @@ public void Accept(long participantId)
...
+	public IReadOnlyCollection<AssessmentValue.Value> GetTopValues()
+	{
+		var uniqueValues = _storyForEstimates.Select(t => t.Value).ToHashSet();
+		
+		if (uniqueValues.Count == 1)
+		{
+		    return uniqueValues.ToArray();
+		}
+
+		return uniqueValues.OrderByDescending(t => (int)t).Take(2).ToArray();
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Domain/Story.cs:207]: You can remove branch 'if'. Always use 'return _storyForEstimates.Select(t => (int)t.Value).OrderByDescending(t => t).Take(2).ToArray();'

File: src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            .AsInt32().Nullable();
+        
+        Delete
+            .Column("accepted")
+            .FromTable("stories")
+            .InSchema("appraiser");
+        
+        Execute.Sql(@"UPDATE appraiser.stories AS s
+                SET total_value = (SELECT MAX(se.value) AS value
+                   FROM appraiser.stories AS ss
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs:24]: You can remove join in subquery.
`SELECT MAX(se.value) AS value
FROM appraiser.story_for_estimates AS se
WHERE s.id = se.story_id
GROUP BY se.story_id`

File: src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                            JOIN appraiser.story_for_estimates AS se ON ss.id = se.story_id
+                   WHERE s.id = ss.id
+                   GROUP BY ss.id)");
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs:27]: Please add description in second parameter `Execute.Sql()`.

File: src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        Delete
+            .Column("total_value")
+            .FromTable("stories")
+            .InSchema("appraiser");
+
+        Create
+            .Column("accepted")
+            .OnTable("stories")
+            .InSchema("appraiser")
+            .AsBoolean()
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs:41]: Add `.Nullable()`

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AcceptEstimate/Services/AcceptEstimateCommandCreator.cs (hunk 1 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AcceptEstimate/Services/AcceptEstimateCommandCreator.cs]: fixed

File: src/Inc.TeamAssistant.Appraiser.DataAccess/StoryReader.cs (hunk 2 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.DataAccess/StoryReader.cs]: fixed

File: src/Inc.TeamAssistant.Appraiser.Domain/Story.cs (hunk 3 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Domain/Story.cs]: fixed

File: src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs (hunk 4 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs]: fixed

File: src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs (hunk 5 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs]: fixed

File: src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs (hunk 6 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Migrations/2024_08_09_0_AddTotalValue.cs:41]: fixed

**Review by dyatlov-a — APPROVED** [review]

**Pull request merged**